platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/NodeId.java

```diff
@@ -103,6 +104,20 @@ public long id() {
         return id;
     }
 
+    /**
+     * Get a NodeId that is offset from this NodeId by the given amount.
+     *
+     * @param offset the amount to offset by
+     * @return the NodeId offset by the given amount
+     */
+    public NodeId getOffset(long offset) {
+        final long newValue = id + offset;
+        if (newValue < LOWEST_NODE_NUMBER) {
+            throw new IllegalArgumentException("the new NodeId must be non-negative");
```

Comment:
This is potentially misleading.
```suggestion
            throw new IllegalArgumentException("the new NodeId, %d, must not be below the minimum value of %d.".formatted(newValue, LOWEST_NODE_NUMBER));
```